src/convertToObject.js

```diff
@@ -6,7 +6,29 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  if (!sourceString.length) {
+    return {};
+  }
+
+  const noEmptyLines = sourceString
+    .split(';')
+    .map((line) => {
+      return line.trim();
+    })
+    .filter((item) => item.length > 0);
+
+  const convertToObj = noEmptyLines.reduce((prev, item) => {
+    const [key, value] = item.split(':');
+
+    const newKey = key.trim();
+    const newValue = value.trim();
+
+    prev[newKey] = newValue;
+
+    return prev;
+  }, {});
```

Comment:
Regarding the [CODE PRACTICE] checklist, it's recommended to avoid using loops and instead use iteration methods. Although `reduce` is an iteration method, the way it's used here is similar to a loop, building the object step by step. Consider if there's a way to use other iteration methods to construct the object more declaratively.